test/test_external.py

```diff
@@ -242,5 +243,64 @@ def test_interface_load_cache_examples(tmp_path):
         )
 
 
+def test_cols_to_rows():
+    assert cols_to_rows({"a": [1, 2, "NaN"], "b": [1, "NaN", 3]}) == (
+        ["a", "b"],
+        [[1, 1], [2, "NaN"], ["NaN", 3]],
+    )
+    assert cols_to_rows({"a": [1, 2, "NaN", 4], "b": [1, "NaN", 3]}) == (
+        ["a", "b"],
+        [[1, 1], [2, "NaN"], ["NaN", 3], [4, None]],
+    )
+    assert cols_to_rows({"a": [1, 2, "NaN"], "b": [1, "NaN", 3, 5]}) == (
+        ["a", "b"],
+        [[1, 1], [2, "NaN"], ["NaN", 3], [None, 5]],
+    )
+    assert cols_to_rows({"a": None, "b": [1, "NaN", 3, 5]}) == (
+        ["a", "b"],
+        [[None, 1], [None, "NaN"], [None, 3], [None, 5]],
+    )
+    assert cols_to_rows({"a": None, "b": None}) == (["a", "b"], [])
+
+
+def check_dataframe(config):
+    input_df = next(
+        c for c in config["components"] if c["props"].get("label", "") == "Input Rows"
+    )
+    assert input_df["props"]["headers"] == ["a", "b"]
+    assert input_df["props"]["row_count"] == (1, "dynamic")
+    assert input_df["props"]["col_count"] == (2, "fixed")
+
+
+def check_dataset(config, readme_examples):
+    # No Examples
+    if not any(readme_examples.values()):
+        assert not any([c for c in config["components"] if c["type"] == "dataset"])
+    else:
+        dataset = next(c for c in config["components"] if c["type"] == "dataset")
+        assert dataset["props"]["samples"] == [
+            [utils.delete_none(cols_to_rows(readme_examples)[1])]
+        ]
+
+
+@pytest.mark.parametrize(
```

Comment:
Should cover most of the weirdness from malformed READMEs. Haven't actually come across a repo with bad README data but it's possible.